shorty.go

```diff
@@ -14,18 +14,49 @@ type Shorty struct {
 
 // New returns a new shorty, machineID can be any unique id to embed in the generated ids.
 func New(store Store, machineID uint32) *Shorty {
-	return &Shorty{
+	s := &Shorty{
 		s:         store,
 		machineID: machineID,
 	}
+	go s.cleanup()
+	return s
+}
+
+func (s *Shorty) cleanup() {
+	for {
```

Comment:
the function should return when the store returns an error (which will happen if it's closed).

I'd like to not add a specific close state since we don't take ownership of the store.